wpilibc/src/main/native/include/frc/kinematics/SwerveModuleState.h

```diff
@@ -25,5 +25,23 @@ struct SwerveModuleState {
    * Angle of the module.
    */
   Rotation2d angle;
+
+  /**
+   * Optimizes the angle of the module to make sure it doesn't
+   * rotate more than 90 degrees from the current state
+   */
+  SwerveModuleState OptimizeModuleAngle(
+      const SwerveModuleState& desiredState) const {
+    SwerveModuleState finalStateAfterOptimization = desiredState;
+    Rotation2d deltaAngle{finalStateAfterOptimization.angle.Degrees() -
+                          angle.Degrees()};
+    if (units::math::abs(deltaAngle.Degrees()) > 90_deg &&
+        units::math::abs(deltaAngle.Degrees()) < 270_deg) {
+      finalStateAfterOptimization.angle =
+          finalStateAfterOptimization.angle + Rotation2d(180_deg);
```

Comment:
Isn't this the same as negating the angle?

Reply:
If by negating you just mean from pos -> neg and neg -> pos, it isn't. If your desired angle is 135 and your starting angle is 0. You would end up with -135 degrees as your set point. This is 90 degrees off from the desired set angle. I have attached a crude paint drawing. 
![swerve_optimization](https://user-images.githubusercontent.com/6174102/72440724-60a0ac80-3777-11ea-830a-42607b0c5f3b.png)

Reply:
135 to -135 degrees is negating, no?

Reply:
if you move your module to -135 you're going to be moving the wrong direction. Think about a swerve drive rotating CCW. All of your modules start at 0. Then you command a full speed CCW rotation. Your wheels will form a "circle" shape. However, if you negated the angles of the wheels that needed to move more than 90 degrees. Your two left wheels would form half of an X shape which wont allow the robot to rotate freely.